choose_visualization.py

```
@@ -0,0 +1,158 @@
+    except Exception as e:
+        logger.error(f"Error in LLM chart selection: {e}")
+        return "none"
+
+
+def _quick_heuristics(question: str, sql: str, rows: list, num_rows: int, num_cols: int) -> str:
+    """
+    Fast heuristic rules to determine chart type without LLM.
+    Returns 'unknown' if heuristics can't decide.
+    """
+    question_lower = question.lower()
+    sql_lower = sql.lower()
```

Ruff (0.12.2)

Local variable `sql_lower` is assigned to but never used

Remove assignment to unused variable `sql_lower`

(F841)